### PR TITLE
Weak Mechanical Soul

### PR DESCRIPTION
**Pull request by SirEdvin**
Hello, so this is new turtle upgrade - Weak Mechanical Soul

Main idea of implementation is give turtle some cool abilities to be like a simple minion. It is weak, because can do only limited things and should consume fuel in process. Main issue is #123 

- [x] Add digBlock function
- [x] Add suckItems function
- [x] Add detectItems function
- [x] Add suckSpecificItem funtion
- [x] Add clickBlock function
- [x] Add fuel consuming logic
- [x] Add ability for turtle to convert RF consuming logic (such as for GeoScanner) to fuel consuming
- [x] Add hunger soul with ability to consume entities and transform to entity soul after enough consumption
- [x] Add End Mechanical Soul that created after 10-20 enderman consumption and will allow to use warp logic.
- [x] Final big test!
- [x] In-game documentation
- [x] Add textures
- [x] Add rate limit with fuel burn rate logic for most operation to avoid overpowered

@Seniorendi maybe you have some suggestion or ideas about this?)

### CODE REVIEW

**Comment by Seniorendi** [comment]
This is incredible! You also added the patchouli documentation. 
The code looks good so far. I will now test this ingame 👍🏻

File: src/main/java/de/srendi/advancedperipherals/common/items/WeakMechanicSoul.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        public final Item resultSoul;
+
+        public MechanicalSoulRecord(int requiredCount, Class<? extends Entity> entityClass, Item resultSoul) {
+            this.requiredCount = requiredCount;
+            this.entityClass = entityClass;
+            this.resultSoul = resultSoul;
+        }
+    }
+
+    // So, I have
```
Comment (Seniorendi) [inline src/main/java/de/srendi/advancedperipherals/common/items/WeakMechanicSoul.java:41]: Mh?

Comment (SirEdvin) [reply src/main/java/de/srendi/advancedperipherals/common/items/WeakMechanicSoul.java]: Oh, will fix now) This is not about code below

**Review by Seniorendi — CHANGES_REQUESTED** [review]
You don't add the turtles to the creative inventory. Take a look at 
https://github.com/SquidDev-CC/CC-Tweaked/blob/7086cb8a02eb030de3f06d514f9ed0a7f7660b7d/src/main/java/dan200/computercraft/shared/turtle/upgrades/TurtleTool.java#L89-L101
and
https://github.com/Seniorendi/AdvancedPeripherals/blob/master/src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/base/BaseTurtle.java#L36-L49

**Comment by SirEdvin** [comment]
@Seniorendi are you sure about creative inventory? I can find turtles in creative inventory. They just have a same texture like a chunky turtle now)

**Comment by Seniorendi** [comment]
> @Seniorendi are you sure about creative inventory? I can find turtles in creative inventory. They just have a same texture like a chunky turtle now)

Oh yeah, I didn't pulled your commits...

**Review by Seniorendi — APPROVED** [review]

**Comment by Seniorendi** [comment]
For any reason, circle ci does not build this PR. But it seems that you are using JDK 11 and not 8. Could you quickly switch to JDK 8 and fix the build issues?

**Comment by SirEdvin** [comment]
Sure, I will do it now) Sorry for this(

**Comment by SirEdvin** [comment]
Fixed)

**Comment by Seniorendi** [comment]
The game crashes when I open the creative tab. It happens when the game tries to render the turtle.
https://gist.github.com/Seniorendi/afac744631f1388e8c181402d835bec0

**Comment by SirEdvin** [comment]
Oh, shame on me, I have forget about creative tab( Thanks! I will fix it right now)

**Comment by SirEdvin** [comment]
Fixed and added `Nullable` annotation to prevent this problem in future (I have no idea that peripheral can be null in some cases( )

**Comment by Seniorendi** [comment]
Ouh because I can't look into the ingame documentation for the new turtles, I need to ask, which advancements do I need to unlock to see this?

**Comment by SirEdvin** [comment]
You need to get Weak Mechanic soul for first pages (about mechanic soul and about consuming logic), End Mechanic soul and Husbandry Mechanic soul for another pages and one of Overpowered Mechanic soul for ovepowered mechanic soul page)

**Comment by Seniorendi** [comment]
Wow, this is amazing!
Tested a bunch of features, this is just amazing. The textures are also very good!
But I have one question, does the turtles now need Energy or not?

**Comment by SirEdvin** [comment]
Turtle doesn't need energy. It need fuel for some operation, if fuel requirements are enabled in configuration. 

Refuel turtle with RF instead of burn items is only one single peripheral method.

**Comment by Seniorendi** [comment]
Kay, then I now understand this. I will now merge this and the documentation PR.
Thank you for the amazing new features!

**Pull request merged**